src/validator.ts

```diff
@@ -17,28 +14,20 @@ const log = logger('ipns:validator')
 const MAX_RECORD_SIZE = 1024 * 10
 
 /**
- * Validates the given ipns entry against the given public key
+ * Validates the given IPNS Record against the given public key. We need a "raw"
+ * record in order to be able to access to all of its fields.
  */
-export const validate = async (publicKey: PublicKey, entry: IPNSEntry): Promise<void> => {
-  const { value, validityType, validity } = entry
+export const validate = async (publicKey: PublicKey, buf: Uint8Array): Promise<void> => {
```

Comment:
We cannot allow the interface here as it abstracts us from the actual protobuf. We need to validate the protobuf.